src/charm.py

```diff
@@ -246,6 +315,46 @@ def _on_oauth_info_changed(self, event: OAuthInfoChangedEvent) -> None:
         """
         self._handle_status_update_config(event)
 
+    @log_event_handler(logger)
+    def _on_invalid_forward_auth_config(self, event: InvalidForwardAuthConfigEvent) -> None:
+        logger.info(
+            "The forward-auth config is invalid: one or more of the related apps is missing ingress relation"
+        )
+        self.unit.status = BlockedStatus(event.error)
+
+    @log_event_handler(logger)
+    def _on_forward_auth_proxy_set(self, event: ForwardAuthProxySet) -> None:
+        logger.info("The proxy was set successfully")
+        self.unit.status = ActiveStatus("OAuth2 Proxy is configured")
+
+    @log_event_handler(logger)
+    def _on_forward_auth_relation_removed(self, event: ForwardAuthRelationRemovedEvent) -> None:
+        logger.info("The proxy was unset")
+        # The proxy was removed, but the charm is still functional
+        self.unit.status = ActiveStatus()
+
+    @log_event_handler(logger)
+    def _on_auth_proxy_config_changed(self, event: AuthProxyConfigChangedEvent) -> None:
+        if not self._oauth2_proxy_service_is_running:
+            self.unit.status = WaitingStatus("Waiting for OAuth2 Proxy service")
+            event.defer()
+            return
+
+        self._handle_status_update_config(event)
+
+        logger.info("Auth-proxy config has changed. Forward-auth relation will be updated")
+        self.forward_auth.update_forward_auth_config(self._forward_auth_config)
```

Comment:
Shouldn't we validate that this is the leader?

Reply:
Leadership is already checked in `forward-auth`, see comment above